lib/matplotlib/tests/test_animation.py

```diff
@@ -282,4 +277,6 @@ def frames_generator():
     anim.save('unused.null', writer=writer)
     assert len(frames_generated) == 5
     for f in frames_generated:
-        assert weakref_assertion_fn(f())
+        # If cache_frame_data is True, then the weakref should be alive;
+        # if cache_frame_data is False, then the weakref should be dead (None).
+        assert (f() is None) != cache_frame_data
```

Comment:
Woundn't it be more clear to add a message to the assert?

Reply:
I think the comments above are good enough, and adding code to generate the correct message would actually obfuscate the issue at hand.